.github/ci-test-each-commit-exec.sh

```diff
@@ -11,15 +11,11 @@ set -o errexit -o pipefail -o xtrace
 
 echo "Running test-one-commit on $( git log -1 )"
 
-git config user.email "ci@example.com"
-git config user.name "CI"
-git merge --no-commit origin/master
-
 # Use clang++, because it is a bit faster and uses less memory than g++
 CC=clang CXX=clang++ cmake -B build -DWERROR=ON -DWITH_ZMQ=ON -DBUILD_GUI=ON -DBUILD_BENCH=ON -DBUILD_FUZZ_BINARY=ON -DWITH_BDB=ON -DWITH_USDT=ON -DCMAKE_CXX_FLAGS='-Wno-error=unused-member-function'
 
 cmake --build build -j "$( nproc )" && ctest --output-on-failure --stop-on-failure --test-dir build -j "$( nproc )"
 
 ./build/test/functional/test_runner.py -j $(( $(nproc) * 2 )) --combinedlogslen=99999999
 
-git merge --abort
+git merge --abort || echo "Nothing to abort. The state was already up to date."
```

Comment:
if we need to keep this change, maybe mention that the abort is related to merging